.github/workflows/rust-checks.yml

```diff
@@ -52,3 +58,24 @@ jobs:
       - name: Run Tests
         run: cargo test
         working-directory: tools/compiler
+
+  execute_and_check:
+    name: Generate registry and Check for Diffs
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: 'true'
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - name: Generate registry
+        run: cargo run
+        env:
+          RUST_LOG: info
+        working-directory: tools/compiler
+      - name: Check for Uncommitted Changes
+        run: git diff --exit-code
+        working-directory: tools/compiler
```

Comment:
Allows us now to ensure folks are properly running the script and not doing manual edits